test/test_rpc.py

```diff
@@ -110,5 +134,74 @@ def test_join_rpc(self):
         # it's safe to call join_rpc() multiple times
         dist.join_rpc()
 
+    @_wrap_with_rpc
+    def test_py_built_in(self):
+        n = self.rank + 1
+        dstRank = n % self.world_size
+        ret = dist.rpc('worker%d' % dstRank, min, args=(n, n + 1, n + 2))
+        self.assertEqual(ret, min(n, n + 1, n + 2))
+
+    @_wrap_with_rpc
+    def test_py_user_defined(self):
+        n = self.rank + 1
+        dstRank = n % self.world_size
+        ret = dist.rpc('worker%d' % dstRank, my_function, kwargs={"a" : n, "b" : n + 1, "c" : n + 2})
+        self.assertEqual(ret, my_function(n, n + 1, n + 2))
+
+    @_wrap_with_rpc
+    def test_py_class_constructor(self):
+        n = self.rank + 1
+        dstRank = n % self.world_size
+        ret = dist.rpc('worker%d' % dstRank, my_class, args=(n,))
+        self.assertEqual(ret.a, n)
+
+    @_wrap_with_rpc
+    def test_py_class_instance_method(self):
+        n = self.rank + 1
+        dstRank = n % self.world_size
+        ret = dist.rpc('worker%d' % dstRank, my_class(2).my_instance_method, args=(n,))
+        self.assertEqual(ret, my_class(2).my_instance_method(n))
+
+    @_wrap_with_rpc
+    def test_py_class_method(self):
+        n = self.rank + 1
+        dstRank = n % self.world_size
+        ret = dist.rpc('worker%d' % dstRank, my_class.my_class_method, args=(n, n + 1))
+        self.assertEqual(ret, my_class.my_class_method(n, n + 1))
+
+    @_wrap_with_rpc
+    def test_py_class_static_method(self):
+        n = self.rank + 1
+        dstRank = n % self.world_size
+        ret = dist.rpc('worker%d' % dstRank, my_class.my_static_method, args=(n + 10,))
+        self.assertEqual(ret, my_class.my_static_method(n + 10))
+
+    @_wrap_with_rpc
+    def test_py_multi_aync_call(self):
+        n = self.rank + 1
+        dstRank = n % self.world_size
+        fut1 = dist.rpc('worker%d' % dstRank, my_class.my_static_method, args=(n + 10,), async_call=True)
+        fut2 = dist.rpc('worker%d' % dstRank, min, args=(n, n + 1, n + 2), async_call=True)
+        self.assertEqual(fut1.wait(), my_class.my_static_method(n + 10))
+        self.assertEqual(fut2.wait(), min(n, n + 1, n + 2))
+
+    @_wrap_with_rpc
+    def test_py_no_return_result(self):
+        n = self.rank + 1
+        dstRank = n % self.world_size
+        ret = dist.rpc('worker%d' % dstRank, no_result)
+        self.assertEqual(ret, no_result())
+
+    @_wrap_with_rpc
+    def test_py_function_exception(self):
+        n = self.rank + 1
+        dstRank = n % self.world_size
+        ret = dist.rpc('worker%d' % dstRank, no_result, args=(10,))
```

Comment:
Please use `.format()`; the percent syntax is not recommended. (Unfortunately, as long as we support Python 3.5 you can't use Python 3 f-strings)